File: mvp/test-fixtures/all-languages/src/services/ProductService.py
from typing import List, Optional
from ..models.Product import Product


class ProductService:
    """Business logic for product operations"""
    
    def __init__(self):
        self.products = {}
        self.next_id = 1
    
    def get_all_products(self) -> List[Product]:
        """Get all products"""
        return list(self.products.values())
    
    def get_product_by_id(self, product_id: int) -> Optional[Product]:
        """Get product by ID"""
        return self.products.get(product_id)
    
    def create_product(self, name: str, price: float, stock: int = 0) -> Product:
        """Create new product"""
        product = Product(self.next_id, name, price, stock)
        self.next_id += 1
        self.products[product.id] = product
        return product
    
    def delete_product(self, product_id: int) -> bool:
        """Delete product"""
        if product_id in self.products:
            del self.products[product_id]
            return True
        return False

File: mvp/test-fixtures/all-languages/src/services/ProductService.js
const Product = require('../models/Product');

class ProductService {
    constructor() {
        this.products = new Map();
        this.nextId = 1;
    }

    getAllProducts() {
        return Array.from(this.products.values());
    }

    getProductById(id) {
        return this.products.get(id);
    }

    createProduct(name, price, stock = 0) {
        const product = new Product(this.nextId++, name, price, stock);
        this.products.set(product.id, product);
        return product;
    }

    deleteProduct(id) {
        return this.products.delete(id);
    }
}

module.exports = ProductService;
